lightrag/lightrag/utils/lazy_import.py

```diff
@@ -1,145 +1,146 @@
-            try:
-                self.module = importlib.import_module(module_path)
-                self.class_ = getattr(self.module, class_name)
-                return self.class_
-            except ImportError:
-                log.info(f"Optional module not installed: {self.import_path}")
-                raise ImportError(f"{self.optional_package.error_message}")
-
-    def __getattr__(self, name):
-        self.load_class()
-        return getattr(self.class_, name)
-
-    def __call__(self, *args, **kwargs):
-        """Create the class instance."""
-        self.load_class()
-        log.debug(f"Creating class instance: {self.class_}")
-        # normal class initialization
-        return self.class_(*args, **kwargs)
-
-
-def safe_import(module_name: str, install_message: str) -> ModuleType:
-    """Safely import a module and raise an ImportError with the install message if the module is not found.
-
-    Mainly used internally to import optional packages only when needed.
-
-    Example:
-
-    1. Tests
-
-    .. code-block:: python
-
-        try:
-            numpy = safe_import("numpy", "Please install numpy with: pip install numpy")
-            print(numpy.__version__)
-        except ImportError as e:
-            print(e)
-
-    When numpy is not installed, it will raise an ImportError with the install message.
-    When numpy is installed, it will print the numpy version.
-
-    2. Use it to delay the import of optional packages in the library.
-
-    .. code-block:: python
-
-        from lightrag.utils.lazy_import import safe_import, OptionalPackages
-
-        numpy = safe_import(OptionalPackages.NUMPY.value[0], OptionalPackages.NUMPY.value[1])
-
-    """
-    try:
-        return importlib.import_module(module_name)
-    except ImportError:
-        raise ImportError(f"{install_message}")
+"""Lazy import a module and class."""
+
+import importlib
+import logging
+from types import ModuleType
+
+
+from enum import Enum
+
+log = logging.getLogger(__name__)
+
+
+class OptionalPackages(Enum):
+    __doc__ = r"""Enum for optional packages that can be used in the library.
+
+    The package name and error message are defined for each optional package as a tuple.
+    """
+    # model sdk
+    GROQ = ("groq", "Please install groq with: pip install groq")
+    OPENAI = ("openai", "Please install openai with: pip install openai")
+    ANTHROPIC = ("anthropic", "Please install anthropic with: pip install anthropic")
+    GOOGLE_GENERATIVEAI = (
+        "google.generativeai",
+        "Please install google-generativeai with: pip install google-generativeai",
+    )
+    TRANSFORMERS = (
+        "transformers",
+        "Please install transformers with: pip install transformers",
+    )
+    COHERE = ("cohere", "Please install cohere with: pip install cohere")
+    OLLAMA = ("ollama", "Please install ollama with: pip install ollama")
+    LITE = ("litellm", "Please install litellm with: pip install litellm")

```

Comment:
Can we modify the name here to be LITELLM. We should also modify the model_client to be litellm_client as well.

Reply:
yes ofc